src/uct/sm/knem/knem_md.c

```diff
@@ -176,30 +227,119 @@ static ucs_status_t uct_knem_rkey_release(uct_md_component_t *mdc, uct_rkey_t rk
     return UCS_OK;
 }
 
-static ucs_status_t uct_knem_md_open(const char *md_name, const uct_md_config_t *md_config,
+static uct_md_ops_t md_ops = {
+    .close             = uct_knem_md_close,
+    .query             = uct_knem_md_query,
+    .mem_alloc         = (void*)ucs_empty_function_return_success,
```

Comment:
since knem does not support UCT_MD_FLAG_ALLOC, we can use this opportunity to delete callbacks for mem_alloc/mem_free

Reply:
didn't get... remove initializer for mem_alloc/free?

Reply:
yes, they will default to NULL